common/json/platform/nlohmann/nlohmann_json.cpp

```diff
@@ -12,24 +12,141 @@
 //    See the License for the specific language governing permissions and
 //    limitations under the License.
 
-#include <iostream>
 #include <common/json.hpp>
+
+#include <fstream>
+#include <string>
 #include <nlohmann/json.hpp>
 
 using njson = nlohmann::json;
+using namespace std;
+
+namespace mender::common::json {
+
+ExpectedJson LoadFromFile(string file_path) {
+	ifstream f(file_path);
+	try {
+		njson parsed = njson::parse(f);
+		Json j = Json(parsed);
+		return ExpectedJson(j);
+	} catch (njson::parse_error &e) {
+		JsonError err = {
+			JsonErrorCode::ParseError, "Failed to parse '" + file_path + "': " + e.what()};
+		return ExpectedJson(err);
+	}
+}
+
+ExpectedJson LoadFromString(string json_str) {
+	try {
+		njson parsed = njson::parse(json_str);
+		Json j = Json(parsed);
+		return ExpectedJson(j);
+	} catch (njson::parse_error &e) {
+		JsonError err = {
+			JsonErrorCode::ParseError, "Failed to parse '''" + json_str + "''': " + e.what()};
+		return ExpectedJson(err);
+	}
+}
+
+string Json::Dump(const int indent) const {
+	return this->n_json.dump(indent);
+}
+
+ExpectedJson Json::Get(const char *child_key) const {
+	bool contains = this->n_json.contains(child_key);
```

Comment:
Hehe, in the end you had to? 🙂

Reply:
Yeah, @oleorhagen pointed me to checking the docs more carefully and they say [this](https://json.nlohmann.me/api/basic_json/operator%5B%5D/#notes):

> If the element with key key does not exist, the behavior is undefined and is guarded by a [runtime assertion](https://json.nlohmann.me/features/assertions/)!